### PR TITLE
Improve personal details window interaction

### PR DESCRIPTION
**Pull request by publicmain**

### CODE REVIEW

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 1]
```diff
@@ -31,124 +35,228 @@ public class PersonDetailsWindow extends UiPart<Stage> {
...
      */
     public PersonDetailsWindow(Logic logic) {
         this(new Stage(), logic);
     }
 
     /**
-     * Creates a new PersonDetailsWindow.
+     * Initializes keyboard event handlers for navigation and actions.
+     */
+    private void initializeKeyHandlers() {
```
Comment (Jalenleeruixian) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:71]: Good use of abstraction with clearly defined fields.

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 2]
```diff
@@ -31,124 +35,228 @@ public class PersonDetailsWindow extends UiPart<Stage> {
...
+            nameField.requestFocus();
+        }
+    }
+
+    /**
+     * Moves focus to the previous {@code TextField} in the form.
+     *
+     * @param current The current TextField.
+     */
+    private void moveFocusToPrevious(TextField current) {
```
Comment (Jalenleeruixian) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:124]: Adequate demonstration of SLAP as the scope of the method is clearly contained within the level of abstraction.

File: src/main/java/seedu/address/ui/PersonListPanel.java [hunk 3]
```diff
@@ -21,21 +22,32 @@ public class PersonListPanel extends UiPart<Region> {
     @FXML
     private ListView<Person> personListView;
     private final PersonDetailsWindow personDetailsWindow;
+
     /**
-     * Creates a {@code PersonListPanel} with the given {@code ObservableList}.
+     * Creates a {@code PersonListPanel} with the given {@code ObservableList} and {@code PersonDetailsWindow}.
```
Comment (Jalenleeruixian) [inline src/main/java/seedu/address/ui/PersonListPanel.java:27]: Good use of Javadocs to make the code usage clearer when reading code.

**Review by Jalenleeruixian — COMMENTED** [review]
Good changes implemented: with a good use of abstraction of methods, well-implemented SLAP, and increased code readability.

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 4]
```diff
@@ -14,11 +16,13 @@
 
 /**
  * Controller for the person details window.
+ * <p>
  * This window allows users to view and edit the details of a selected person.
- * It provides editable text fields for the person's name, email, phone number, and address,
- * along with "Save" and "Cancel" buttons to handle user actions.
```
Comment (WuJinhan1) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:19]: LGTM!

**Review by WuJinhan1 — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 5]
```diff
@@ -31,124 +35,229 @@ public class PersonDetailsWindow extends UiPart<Stage> {
...
      */
     public void show(Person person) {
-        logger.fine("Showing detailed page about this person");
+        logger.fine("Showing detailed page about this person.");
         this.person = person;
-        nameField.setText(person.getName().fullName);
-        emailField.setText(person.getEmail().value);
-        phoneField.setText(person.getPhone().value);
-        addressField.setText(person.getAddress().value);
+        populateFields(person);
```
Comment (WuJinhan1) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:144]: Good job with the code!

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 6]
```diff
@@ -31,124 +35,229 @@ public class PersonDetailsWindow extends UiPart<Stage> {
...
 
             getRoot().hide();
 
         } catch (CommandException | ParseException e) {
             logger.severe("Failed to edit person: " + e.getMessage());
-            Alert alert = new Alert(Alert.AlertType.ERROR);
-            alert.setTitle("Error");
-            alert.setHeaderText("Failed to save changes.");
-            alert.setContentText(e.getMessage());
-            alert.showAndWait();
```
Comment (WuJinhan1) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:124]: I like the change here that you did!

File: src/main/java/seedu/address/ui/PersonListPanel.java (hunk 3 above)
Comment (WuJinhan1) [inline src/main/java/seedu/address/ui/PersonListPanel.java:27]: Java doc is written very nicely here!

File: src/main/resources/view/PersonDetailsWindow.fxml [hunk 7]
```diff
@@ -35,14 +33,6 @@
...
                            style="-fx-text-fill: white;"/>
                     <TextField fx:id="addressField" GridPane.rowIndex="3" GridPane.columnIndex="1"/>
-
-                    <!-- Save and Cancel Buttons -->
-                    <HBox spacing="10" alignment="CENTER_RIGHT" GridPane.rowIndex="4" GridPane.columnIndex="1">
-                        <children>
-                            <Button text="Save" onAction="#handleSave"/>
-                            <Button text="Cancel" onAction="#handleCancel"/>
-                        </children>
-                    </HBox>
```
Comment (WuJinhan1) [inline src/main/resources/view/PersonDetailsWindow.fxml:45]: I like how you changed up the fxml file here!

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 8]
```diff
@@ -5,6 +5,8 @@
 import javafx.fxml.FXML;
 import javafx.scene.control.Alert;
 import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
```
Comment (WuJinhan1) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:9]: LGTM!

File: src/main/java/seedu/address/ui/PersonDetailsWindow.java [hunk 9]
```diff
@@ -31,124 +35,229 @@ public class PersonDetailsWindow extends UiPart<Stage> {
...
+                moveFocusToPrevious(textField);
+                event.consume();
+            } else if (event.getCode() == KeyCode.ESCAPE) {
+                handleCancel();
+                event.consume();
+            } else if (event.getCode() == KeyCode.ENTER) {
+                handleSave();
+                event.consume();
+            }
+        });
```
Comment (testing1234567891011121314) [inline src/main/java/seedu/address/ui/PersonDetailsWindow.java:99]: Good use of SLAP! LGTM

**Pull request merged**